confirms.go

```diff
@@ -168,13 +169,13 @@ func (d *deferredConfirmations) Close() {
 
 	for k, v := range d.confirmations {
 		v.confirmation = Confirmation{DeliveryTag: k, Ack: false}
-		v.wg.Done()
+		v.cancel()
 		delete(d.confirmations, k)
 	}
 }
 
 // Waits for publisher confirmation. Returns true if server successfully received the publishing.
 func (d *DeferredConfirmation) Wait() bool {
-	d.wg.Wait()
+	<-d.ctx.Done()
```

Comment:
If the user canceled the context that he passed to `PublishWithDeferredConfirmWithContext` will this return true aka "server successfully received the publishing"?

If yes, is this correct?
If the user cancelled the context, we don't know if the publishing was received by the server, isn't returning `true` here misleading?

Reply:
If the context is cancel by the user, the default value of `d.confirmation.Ack` would be false. Since we didn't received confirmation yet setting it to true or false [here](https://github.com/rabbitmq/amqp091-go/pull/96/files#diff-baa074956b5a90bfc21c6817a6053635a7958696dbad9fe6969abdeaab19e6f7R158).

I added those tests [with cancelled context](https://github.com/rabbitmq/amqp091-go/pull/96/files#diff-bedc9577ef7b88edae0964a118babc360b4b75aa4ddb9998745656728d5190ddR263) and [with timeout on context](https://github.com/rabbitmq/amqp091-go/pull/96/files#diff-bedc9577ef7b88edae0964a118babc360b4b75aa4ddb9998745656728d5190ddR284) to validate that we receive false on all Wait calls in case of context cancelled.

Reply:
Thanks for the clarification.
I got confused somehow, not sure how I got the impression that it returns `true` here :-)